crates/bevy_tasks/src/task_pool.rs

```diff
@@ -164,26 +190,38 @@ impl TaskPool {
 
     /// Return the number of threads owned by the task pool
     pub fn thread_num(&self) -> usize {
-        self.inner.threads.len()
+        self.thread_count_for(TaskGroup::Compute)
+            + self.thread_count_for(TaskGroup::AsyncCompute)
+            + self.thread_count_for(TaskGroup::IO)
+    }
+
+    /// Return the number of threads owned by a given group in the task pool
+    pub fn thread_count_for(&self, group: TaskGroup) -> usize {
+        self.groups.get(group).threads
     }
 
-    /// Allows spawning non-`'static` futures on the thread pool. The function takes a callback,
-    /// passing a scope object into it. The scope object provided to the callback can be used
-    /// to spawn tasks. This function will await the completion of all tasks before returning.
+    /// Allows spawning non-`'static` futures on the thread pool in a specific task group. The
+    /// function takes a callback, passing a scope object into it. The scope object provided
+    /// to the callback can be used to spawn tasks. This function will await the completion of
+    /// all tasks before returning.
     ///
     /// This is similar to `rayon::scope` and `crossbeam::scope`
-    pub fn scope<'scope, F, T>(&self, f: F) -> Vec<T>
+    pub fn scope<'scope, F, T>(&self, group: TaskGroup, f: F) -> Vec<T>
     where
         F: FnOnce(&mut Scope<'scope, T>) + 'scope + Send,
         T: Send + 'static,
     {
+        if self.thread_count_for(group) == 0 {
```

Comment:
this check doesn't account for the fact that compute tasks can be run on the io and async task groups.

Reply:
Fixed.